docarray/array/storage/base/helper.py

```diff
@@ -2,44 +2,61 @@
 
 
 class Offset2ID:
-    def __init__(self, ids=None):
+    def __init__(self, ids=None, enable_offset2id=None):
         self.ids = ids or []
+        if not enable_offset2id:
+            self._enable_offset2id = False
+        else:
+            self._enable_offset2id = True
 
     def get_id(self, idx):
+        if not self._enable_offset2id:
+            raise ValueError("Offset2id is disabled")
         return self.ids[idx]
 
     def append(self, data):
-        self.ids.append(data)
+        if self._enable_offset2id:
+            self.ids.append(data)
 
     def extend(self, data):
-        self.ids.extend(data)
+        if self._enable_offset2id:
+            self.ids.extend(data)
 
     def update(self, position, data_id):
-        self.ids[position] = data_id
+        if self._enable_offset2id:
+            self.ids[position] = data_id
 
     def delete_by_id(self, _id):
-        del self.ids[self.ids.index(_id)]
+        if self._enable_offset2id:
+            del self.ids[self.ids.index(_id)]
 
     def index(self, _id):
+        if not self._enable_offset2id:
+            raise ValueError("Offset2id is disabled")
```

Comment:
I think we should offer more details information about this error. Maybe we can give some instructions to let the user how to deal with it. For example,
```
raise ValueError('The offset2id is not enable for list-like index. To disable this error, configure the `enable_offset2id` to True')```

Reply:
Make sense.